File: BackEnd/app/shemas/prediction.py
from pydantic import BaseModel


class PredictionRequest(BaseModel):
    Balcony: float
    Bathroom: float
    Super_Area: float
    Carpet_Area: float

    location: str
    Status: str
    Ownership: str
    facing: str
    Transaction: str
    Furnishing: str


class PredictionResponse(BaseModel):
    predicted_price: float

File: FrontEnd/src/types/prediction.ts
export interface PredictionRequest {
    Balcony: number;
    Bathroom: number;
    Super_Area: number;
    Carpet_Area: number;

    location: string;
    Status: string;
    Ownership: string;
    facing: string;
    Transaction: string;
    Furnishing: string;
}

export interface PredictionResponse {
    predicted_price: number;
}